packages/titus-backend/lib/plugins/auth/authzcheck.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const { Forbidden } = require('http-errors')
+const jwt = require('jsonwebtoken')
+
+async function authRoutes(server, options) {
+  // add users to the admin role
+  server.addHook('onReady', async function () {
+    const adminUsers = process.env.CHECK_AUTHZ_ADMIN_USERS || ''
+    const policies = adminUsers
+      .split(',')
+      .map(u => server.casbin.addRoleForUser(u, 'role_admin'))
+    await Promise.all(policies)
+  })
+
+  // this is a sample to check the signed-in user's permission to access this resource via casbin policies based on configuration in the OAuth provider
+  // a real-world example would have more context around the user account and roles/scopes/permissions and policies
+  server.route({
+    method: 'GET',
+    url: '/authzcheck',
+    schema: {
+      tags: ['authz'],
+      security: [
+        {
+          apiKey: []
+        }
+      ]
+    },
+    onRequest: [
+      server.authenticate,
+      async (request, res) => {
+        const authzId = request.headers['x-authz-id']
+        const authzUserData = jwt.decode(authzId)
```

Comment:
didn't we have the user already decoded somewhere? We should try avoiding decoding twice. I would prefer verifying vs decoding (if possible).

Reply:
I believe, but I'll let Adam explain, that this is a different token than the one we already used. What we used is an access token and this one is an id token.

Reply:
Exactly - the `accessToken` doesn't always contain any useful information for assigning roles - but all the auth providers all also define `idToken` with more user info like email